src/views/ManageList.jsx

```diff
@@ -1,7 +1,126 @@
-export function ManageList() {
+import { useState } from 'react';
+import { addItem } from '../api/firebase';
+import toast, { Toaster } from 'react-hot-toast';
+
+const purchaseTimelines = {
+	soon: 7,
+	kindOfSoon: 14,
+	notSoon: 30,
+};
+
+export function ManageList({ listPath }) {
+	const [itemName, setItemName] = useState('');
+	const [itemNextPurchaseTimeline, setItemNextPurchaseTimeline] = useState('');
+
+	const handleItemNameTextChange = (e) => {
+		setItemName(e.target.value);
+	};
+
+	const handleNextPurchaseChange = (e) => {
+		setItemNextPurchaseTimeline(e.target.value);
+	};
+
+	const handleSubmit = async (e) => {
+		e.preventDefault();
+
+		const daysUntilNextPurchase = purchaseTimelines[itemNextPurchaseTimeline];
```

Comment:
since JavaScript is so loosy goosy, we probably should be checking if `purchaseTimelines[itemNextPurchaseTimeline]` exists before continuing, or at least verify that our database won't be put into a bad place if `purchaseTimelines[itemNextPurchaseTimeline]` returns `undefined`

```
const times = {
  "now": 0,
  "sooner": 3,
  "soon": 7,
}

const time = times["now"];

const typo_time = times["yesterday"];

console.log(time, typo_time);
```

We want to do this because we if accidentally let this slip by now, we'd have a problem where our database contains data the client probably doesn't exist; which would lead to errors down the line. If `daysUntilNextPurchase` is undefined, and the database lets it be inserted (assuming that will succeed, worth checking!) when it comes time to render our list, we may rightfully assume a day is stored in that field, try to parse it, and get an error. Now image the person that creates this bad list shares it with other users, now they're impacted too 💣 Normally you'd expect the server to do this kind of validation, but I'm not sure how firebase will behave here.

Reply:
Using a set of constants here can also cut down on the chance of having typos like this. You can declare your times at the top of the file and then reference them in `purchaseTimelines`

```
const SOON = "soon";
const KIND_OF_SOON = "kindOfSoon";
const NOT_SOON = "notSoon"

const purchaseTimelines = {
  [SOON]: 7,
  ... etc
};

# In your render function 

... snip
checked={itemNextPurchaseTimeline === SOON}
... snip
```

Reply:
I know we don't use TypeScript here, but it has native support for enums, which are also helpful when it comes to definitions that get reused - possibly something to 👀 into at some point

Reply:
100%!